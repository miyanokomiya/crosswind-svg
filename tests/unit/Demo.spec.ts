import { mount } from "@vue/test-utils";
import Demo from "../../demo/Demo.vue";

describe("Demo", () => {
  it("default", () => {
    const wrapper = mount(Demo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
