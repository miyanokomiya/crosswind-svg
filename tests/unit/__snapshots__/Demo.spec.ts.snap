// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo default 1`] = `
<div>
  <section>
    <h2>
      OpenCloseIcon
    </h2>
     
    <div>
      <svg
        class="icon"
        style="transform: rotate(0); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <circle
            cx="32"
            cy="32"
            r="32"
          />
           
          <!---->
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 32 20 32 44"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
        </g>
      </svg>
      <svg
        class="icon"
        style="transform: rotate(45deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill=""
        >
          <circle
            cx="32"
            cy="32"
            r="32"
          />
           
          <!---->
        </g>
         
        <g
          stroke="white"
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 32 20 32 44"
            style="transform-origin: 50% 50%; transition: all 1s;"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 1s;"
          />
        </g>
      </svg>
      <svg
        class="icon red"
        style="transform: rotate(180deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <!---->
           
          <!---->
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 32 20 32 44"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
        </g>
      </svg>
      <svg
        class="icon red"
        style="transform: rotate(180deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <!---->
           
          <rect
            height="64"
            rx="6"
            ry="6"
            width="64"
            x="0"
            y="0"
          />
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 32 20 32 44"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
        </g>
      </svg>
    </div>
  </section>
   
  <section>
    <h2>
      HamburgerArrowIcon
    </h2>
     
    <div>
      <svg
        class="icon"
        style="transform: rotate(0); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <circle
            cx="32"
            cy="32"
            r="32"
          />
           
          <!---->
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 20 20 44 20"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 44 44 44"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
        </g>
      </svg>
      <svg
        class="icon"
        style="transform: rotate(45deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill=""
        >
          <circle
            cx="32"
            cy="32"
            r="32"
          />
           
          <!---->
        </g>
         
        <g
          stroke="white"
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 20 20 44 20"
            style="transform-origin: 50% 50%; transition: all 1s;"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 1s;"
          />
           
          <path
            d="M 20 44 44 44"
            style="transform-origin: 50% 50%; transition: all 1s;"
          />
        </g>
      </svg>
      <svg
        class="icon red"
        style="transform: rotate(180deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <!---->
           
          <!---->
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 20 20 44 20"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 44 44 44"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
        </g>
      </svg>
      <svg
        class="icon red"
        style="transform: rotate(180deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <!---->
           
          <rect
            height="64"
            rx="6"
            ry="6"
            width="64"
            x="0"
            y="0"
          />
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 20 20 44 20"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
           
          <path
            d="M 20 44 44 44"
            style="transform-origin: 50% 50%; transition: all 0.5s;"
          />
        </g>
      </svg>
    </div>
  </section>
   
  <section>
    <h2>
      DirectionArrowIcon
    </h2>
     
    <div>
      <svg
        class="icon"
        style="transform: rotate(0); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <circle
            cx="32"
            cy="32"
            r="32"
          />
           
          <!---->
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 20 20 44 20"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(45deg) scaleX(-1);"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(90deg);"
          />
           
          <path
            d="M 20 44 44 44"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-45deg) scaleY(-1);"
          />
        </g>
      </svg>
      <svg
        class="icon"
        style="transform: rotate(45deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill=""
        >
          <circle
            cx="32"
            cy="32"
            r="32"
          />
           
          <!---->
        </g>
         
        <g
          stroke="white"
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 20 20 44 20"
            style="transform-origin: 50% 50%; transition: all 1s; transform: rotate(45deg) scaleX(-1);"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 1s; transform: rotate(90deg);"
          />
           
          <path
            d="M 20 44 44 44"
            style="transform-origin: 50% 50%; transition: all 1s; transform: rotate(-45deg) scaleY(-1);"
          />
        </g>
      </svg>
      <svg
        class="icon red"
        style="transform: rotate(180deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <!---->
           
          <!---->
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 20 20 44 20"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(45deg) scaleX(-1);"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(90deg);"
          />
           
          <path
            d="M 20 44 44 44"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-45deg) scaleY(-1);"
          />
        </g>
      </svg>
      <svg
        class="icon red"
        style="transform: rotate(180deg); fill: currentColor; stroke: currentColor;"
        viewBox="0 0 64 64"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <g
          fill="white"
        >
          <!---->
           
          <rect
            height="64"
            rx="6"
            ry="6"
            width="64"
            x="0"
            y="0"
          />
        </g>
         
        <g
          stroke=""
          stroke-linecap="round"
          stroke-width="4"
        >
          <path
            d="M 20 20 44 20"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(45deg) scaleX(-1);"
          />
           
          <path
            d="M 20 32 44 32"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(90deg);"
          />
           
          <path
            d="M 20 44 44 44"
            style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-45deg) scaleY(-1);"
          />
        </g>
      </svg>
    </div>
  </section>
</div>
`;
